sdcm/utils/sstable/sstable_utils.py

```diff
@@ -243,36 +234,50 @@ def _are_tombstones_in_sstabledump(self, sstable: str, remote_json_path: str = R
         self.log.debug("Tombstones are found in SSTable %s.", sstable)
         return True
 
-    def get_compacted_tombstone_deletion_info(self, sstable: str) -> list:
+    def _get_sstable_dump_data(self, sstable: str) -> dict | None:
         """
-        Extracts tombstone deletion info from a compacted SSTable dump.
+        Extracts Json data from a compacted SSTable dump.
 
         :param sstable: The SSTable file path.
-        :return: List of tombstone deletion entries.
+        :return: sstable data in a Json format.
         """
         if not self._run_sstabledump(sstable=sstable):  # Check if SSTable exists and was dumped
             self.log.debug("Skipping tombstone search as SSTable %s does not exist or dump failed.", sstable)
-            return []
+            return None
         if not self._are_tombstones_in_sstabledump(sstable=sstable):
-            return []
+            return None
 
-        tombstones_deletion_info = []
         result = self.db_node.remoter.run(f'sudo cat {self.REMOTE_SSTABLEDUMP_PATH}', verbose=False,
                                           ignore_status=False)
 
         if not result.ok:
             self.log.warning("Failed to retrieve SSTable dump data for %s: (%s, %s)", sstable, result.stdout,
                              result.stderr)
-            return tombstones_deletion_info
+            return None
 
         try:
-            dump_data = json.loads(result.stdout)
+            return json.loads(result.stdout)
 
-            # Get the list of records for the given SSTable
-            sstable_data = dump_data['sstables'].get(sstable, [])
+        except json.JSONDecodeError as e:
+            self.log.error("Failed to parse SSTable dump JSON for %s: %s", sstable, str(e))
+            raise
 
-            # Extract entries that contain a tombstone
-            tombstones_deletion_info = [entry for entry in sstable_data if 'tombstone' in entry]
+    def get_compacted_tombstone_deletion_info(self, sstable: str) -> list:
+        """
+        Extracts tombstone deletion info from a compacted SSTable dump.
+
+        :param sstable: The SSTable file path.
+        :return: List of tombstone deletion entries.
+        """
+        tombstones_deletion_info = []
+        try:
+            if dump_data := self._get_sstable_dump_data(sstable=sstable):
+
+                # Get the list of records for the given SSTable
+                sstable_data = dump_data['sstables'].get(sstable, [])
+
+                # Extract entries that contain a tombstone
+                tombstones_deletion_info = [entry for entry in sstable_data if 'tombstone' in entry]
```

Comment:
This will match only the partition tombstone.
Is this intended?

Reply:
yes, with this regard, SCT, so far, only handles partition tombstone.
Yet it is a good point and meant to be addressed in tombstone test plan and in tests.
I opened an issue for that - https://github.com/scylladb/scylla-cluster-tests/issues/10595.